www/resources/views/pages/breakdownTimeline.blade.php

```diff
@@ -0,0 +1,184 @@
+// found in the LICENSE.md file.
+
+// TODO: use these in the template
+// $page_keywords = array('Timeline Breakdown','WebPageTest','Website Speed Test','Page Speed');
+// $page_description = "Chrome main-thread processing breakdown$testLabel";
+
+?>
+<div class="results_main_contain">
+    <div class="results_main">
+       <div class="results_and_command">
+          <div class="results_header">
+             <h2>Main-thread Processing</h2>
+             <p>Where the browser's main thread was busy, not including idle time waiting for resources
+                <a href={{ $timeline_url }} title="View Chrome Dev Tools Timeline">view timeline</a>
+                .
+             </p>
+          </div>
+       </div>
+       <div id="result" class="results_body">
+          <h3 class="hed_sub">Processing Breakdown</h3>
+          <div class="breakdownFrame">
+             <div class="breakdownFrame_item">
+                <div id="pieGroups" ></div>
+                <div class="table visualization_table" id="tableGroups" ></div>
+             </div>
+             <div class="breakdownFrame_item">
+                <div id="pieEvents" ></div>
+                <div class="table visualization_table" id="tableEvents"></div>
+             </div>
+          </div>
+          <h3 class="hed_sub">Timing Breakdown</h3>
+          <p>All of the main-thread activity including idle (waiting for resources usually)
+             <a href={{ $timeline_url }} title="View Chrome Dev Tools Timeline">view timeline</a>
+          </p>
+          <div class="breakdownFrame">
+             <div class="breakdownFrame_item">
+                <div id="pieGroupsIdle" ></div>
+                <div class="table visualization_table" id="tableGroupsIdle"></div>
+             </div>
+             <div class="breakdownFrame_item">
+                <div id="pieEventsIdle"></div>
+                <div class="table visualization_table" id="tableEventsIdle"></div>
+             </div>
+          </div>
+       </div>
+    </div>
+ </div>
+
+<!--Load the AJAX API-->
+<script src="//www.google.com/jsapi"></script>
+<script>
+   // Load the Visualization API and the table package.
+   google.load('visualization', '1', {'packages':['table', 'corechart']});
+   google.setOnLoadCallback(drawTable);
+
+   function drawTable() {
+       var groups = new google.visualization.DataTable();
+       groups.addColumn('string', 'Category');
+       groups.addColumn('number', 'Time (ms)');
+       groups.addRows(<?php echo (count($groups) - 1); ?>);
+       var groupColors = new Array();
+   
+       var events = new google.visualization.DataTable();
+       events.addColumn('string', 'Event');
+       events.addColumn('number', 'Time (ms)');
+       events.addRows(<?php echo (count($processing) - 1); ?>);
+       var eventColors = new Array();
+       <?php
+        $index = 0;
+        if (isset($groups) && is_array($groups) && count($groups)) {
+            foreach ($groups as $type => $time) {
```

Comment:
there are loops and ifs in Blade, could you try them?

https://laravel.com/docs/9.x/blade#if-statements
https://laravel.com/docs/9.x/blade#loops

the idea is to minimize `<?php` as much as we can

Reply:
And `echo` is really what we don't need when we have templates :)